opentracing-brave/src/main/java/io/opentracing/BraveSpanImpl.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.ServerTracer;
+import com.github.kristofa.brave.SpanId;
+
+
+final class BraveSpanImpl extends AbstractSpan {
+
+    final SpanId spanId;
+    final Optional<Span> parent;
+    final Optional<ServerTracer> serverTracer;
```

Comment:
The serverTracer and clientTracer are only ever inserted here if this span was involved in or created by calls to `tracer.inject(..)` and `tracer.extract(..)`.

When you call `tracer.inject(..)` you annotate a CS and then when `span.finish()` is called the span must annotate a CR, hence in this situation the `tracer.inject(..)` process gives the span the clientTracer reference.

A similare thing with `tracer.extract(..)`, the SR and SS annotations, and the serverTracer reference here.

I don't yet see how manually creating spans will simplify the code. Maybe an example?